### ISSUE TITLE
Error building description-extractor during installation

### ISSUE THREAD

**Issue by TimD1** [mutalyzer/mutalyzer#101]
I am unable to install `mutalyzer` (both through `pip` and from source) on both Linux and MacOSX (intel emulation), due to a failed `description-extractor` build. Do you have any suggestions for how I can successfully install mutalyzer?

```
  Building wheel for description-extractor (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [54 lines of output]
      running bdist_wheel
      running build
      running build_ext
      Cloning into 'extractor-core'...
      building 'extractor' extension
      creating build
      creating build/temp.linux-x86_64-pypy39
      creating build/temp.linux-x86_64-pypy39/extractor-core
      creating build/temp.linux-x86_64-pypy39/extractor-core/src
      /cluster/home/dunnt/env/mambaforge/bin/x86_64-conda-linux-gnu-cc -DNDEBUG -O2 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /cluster/home/dunnt/env/mambaforge/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /cluster/home/dunnt/env/mambaforge/include -fPIC -I/cluster/home/dunnt/env/mambaforge/include/pypy3.9 -c extractor-core/src/extractor.cc -o build/temp.linux-x86_64-pypy39/extractor-core/src/extractor.o -Wextra -Wpedantic -std=c++17
      extractor-core/src/extractor.cc: In function 'size_t mutalyzer::LCS_1(std::vector<Substring>&, const char_t*, const char_t*, size_t, size_t, const char_t*, size_t, size_t)':
      extractor-core/src/extractor.cc:1015:18: warning: ISO C++ forbids variable length array 'array' [-Wvla]
       1015 |   typedef size_t array[2][reference_length];
            |                  ^~~~~
      extractor-core/src/extractor.cc: In function 'size_t mutalyzer::LCS_k(std::vector<Substring>&, const char_t*, const char_t*, size_t, size_t, const char_t*, size_t, size_t, size_t)':
      extractor-core/src/extractor.cc:1130:18: warning: ISO C++ forbids variable length array 'array' [-Wvla]
       1130 |   typedef size_t array[k + 1][reference_length];
            |                  ^~~~~
      extractor-core/src/extractor.cc:1130:18: warning: ISO C++ forbids variable length array 'array' [-Wvla]
      extractor-core/src/extractor.cc: In function 'void mutalyzer::LCS_frame_shift(std::vector<Substring>&, const char_t*, size_t, size_t, const char_t*, size_t, size_t)':
      extractor-core/src/extractor.cc:1335:10: warning: ISO C++ forbids variable length array 'lcs' [-Wvla]
       1335 |   size_t lcs[2][reference_length][5];
            |          ^~~
      extractor-core/src/extractor.cc: In function 'void mutalyzer::backtranslation(char_t*, char_t*, const char_t*, size_t, const char_t*, size_t, size_t, uint8_t)':
      extractor-core/src/extractor.cc:1553:10: warning: ISO C++ forbids variable length array 'reference_DNA' [-Wvla]
       1553 |   size_t reference_DNA[3 * length];
            |          ^~~~~~~~~~~~~
      extractor-core/src/extractor.cc:1554:10: warning: ISO C++ forbids variable length array 'sample_DNA' [-Wvla]
       1554 |   size_t sample_DNA[3 * length];
            |          ^~~~~~~~~~
      /cluster/home/dunnt/env/mambaforge/bin/x86_64-conda-linux-gnu-cc -DNDEBUG -O2 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /cluster/home/dunnt/env/mambaforge/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /cluster/home/dunnt/env/mambaforge/include -fPIC -I/cluster/home/dunnt/env/mambaforge/include/pypy3.9 -c extractor-wrapper.cc -o build/temp.linux-x86_64-pypy39/extractor-wrapper.o -Wextra -Wpedantic -std=c++17
      extractor-wrapper.cc:259:1: warning: missing initializer for member 'PyModuleDef::m_slots' [-Wmissing-field-initializers]
        259 | }; // extractormodule
            | ^
      extractor-wrapper.cc:259:1: warning: missing initializer for member 'PyModuleDef::m_traverse' [-Wmissing-field-initializers]
      extractor-wrapper.cc:259:1: warning: missing initializer for member 'PyModuleDef::m_clear' [-Wmissing-field-initializers]
      extractor-wrapper.cc:259:1: warning: missing initializer for member 'PyModuleDef::m_free' [-Wmissing-field-initializers]
      extractor-wrapper.cc: In function 'int main(int, char**)':
      extractor-wrapper.cc:276:30: error: 'Py_DecodeLocale' was not declared in this scope; did you mean 'PyUnicode_DecodeLocale'?
        276 |     wchar_t* const program = Py_DecodeLocale(argv[0], NULL);
            |                              ^~~~~~~~~~~~~~~
            |                              PyUnicode_DecodeLocale
      extractor-wrapper.cc:283:5: error: 'PyImport_AppendInittab' was not declared in this scope
        283 |     PyImport_AppendInittab("extractor", PyInit_extractor);
            |     ^~~~~~~~~~~~~~~~~~~~~~
      extractor-wrapper.cc:285:5: error: 'Py_SetProgramName' was not declared in this scope; did you mean 'Py_GetProgramName'?
        285 |     Py_SetProgramName(program);
            |     ^~~~~~~~~~~~~~~~~
            |     Py_GetProgramName
      extractor-wrapper.cc:287:5: error: 'Py_Initialize' was not declared in this scope; did you mean 'Py_IsInitialized'?
        287 |     Py_Initialize();
            |     ^~~~~~~~~~~~~
            |     Py_IsInitialized
      error: command '/cluster/home/dunnt/env/mambaforge/bin/x86_64-conda-linux-gnu-cc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for description-extractor
  Running setup.py clean for description-extractor
Successfully built mutalyzer
Failed to build description-extractor
ERROR: Could not build wheels for description-extractor, which is required to install pyproject.toml-based projects
```

**Comment by mihailefter** [comment]
@TimD1: Could you provide more information regarding your environment?

**Comment by TimD1** [comment]
This was also replicatable in a Docker environment:
```
> docker pull --platform linux/amd64 ubuntu:24.04 
> docker run --platform linux/amd64 -it ubuntu:24.04
$ apt-get update
$ apt-get install python3 python3-pip python3.12-venv
$ python3 -m venv mut
$ source ./mut/bin/activate
(mut) $ pip install mutalyzer
```
Which resulted in:
```
Building wheels for collected packages: description-extractor, mutalyzer-algebra, mutalyzer-backtranslate, mutalyzer-crossmapper
  Building wheel for description-extractor (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for description-extractor (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [4 lines of output]
      running bdist_wheel
      running build
      running build_ext
      error: [Errno 2] No such file or directory: 'git'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for description-extractor
  Building wheel for mutalyzer-algebra (pyproject.toml) ... done
  Created wheel for mutalyzer-algebra: filename=mutalyzer_algebra-1.5.1-py3-none-any.whl size=25264 sha256=d4564799b0dcbaf9d8615816ab77f71f3fc510ba3c838b12f632897e57be8e3d
  Stored in directory: /root/.cache/pip/wheels/1a/f7/d4/e5f900644766926cbb583cc3971bcb62508648faab41aa5361
  Building wheel for mutalyzer-backtranslate (pyproject.toml) ... done
  Created wheel for mutalyzer-backtranslate: filename=mutalyzer_backtranslate-1.0.0-py3-none-any.whl size=6829 sha256=5fab2f0d18703725d59d28cd6ec28539a56ab928d1c9e53214bfb09ca448049a
  Stored in directory: /root/.cache/pip/wheels/ed/21/62/b5e9b76c0042c20355d8cea09528ac430374bca8c3a6e64c22
  Building wheel for mutalyzer-crossmapper (pyproject.toml) ... done
  Created wheel for mutalyzer-crossmapper: filename=mutalyzer_crossmapper-2.0.1-py3-none-any.whl size=6526 sha256=88c04fa02abd4e6af484c8ab8c0fab1bb050f071a69103abfc8ad48e9d6a20a9
  Stored in directory: /root/.cache/pip/wheels/79/91/99/17cb1086352227a46d86d7848d68c398b207b980bd74cb8510
Successfully built mutalyzer-algebra mutalyzer-backtranslate mutalyzer-crossmapper
Failed to build description-extractor
ERROR: Could not build wheels for description-extractor, which is required to install pyproject.toml-based projects
```

**Comment by mihailefter** [comment]
@TimD1: Regarding your email about docker, I notice this:

`error: [Errno 2] No such file or directory: 'git'`

Make sure that you have `git` installed in the container.

Also, reply using the GitHub.

**Comment by TimD1** [comment]
Thanks for the quick responses! The email originated from my reply on Github, which I deleted once I noticed the same issue you pointed out. I was able to successfully build `mutalyzer` in both a Docker and conda environment.

My issue seems to have been that `mutalyzer` depends on `description-extractor`, which `pip` apparently needs to build from source. Installing through `pip` therefore has a few hidden dependencies, including `g++` (not all versions work) and `git`.

Thanks for all your work on this great project! It solved my need for a tool that can do HGVS conversion with a custom reference, and no external database connections.
